code/controllers/subsystem/input.dm

```diff
@@ -36,4 +36,6 @@ SUBSYSTEM_DEF(input)
 
 /datum/controller/subsystem/input/fire()
 	for(var/mob/user as anything in GLOB.player_list)
+		if(SSlag_switch.measures[DISABLE_DEAD_KEYLOOP] && user.stat == DEAD && !user.client?.holder)
```

Comment:
wouldn't it be more effective to bookkeep a "keyloop" list and add/remove users as they are eligible for the keyloop.

Sure, you save the call into the keyloop procs but right now you still have to pay the cost of iterating

Reply:
I hadn't considered that, but now that you mention it I'm leaning that way too since we'd just need to add everyone to the list by default most of the time and then prevent their adding later if the measure is in place and prune for non-staff when it's enabled then re-add the whole player_list if turned back off.